src/gateway.rs

```diff
@@ -173,18 +182,30 @@ impl Gateway {
         }
     }
 
-    async fn handle_transmit_beacon(&mut self, logger: &Logger, beacon: Beacon) {
+    fn tx_power(&mut self, logger: &Logger) -> Option<u32> {
         let region_params = if let Some(region_params) = &self.region_params {
             region_params
         } else {
             warn!(logger, "ignoring transmit request, no region params");
-            return;
+            return None;
         };
 
-        let tx_power = if let Some(tx_power) = region_params.tx_power() {
-            tx_power
+        if let Some(tx_power) = region_params.tx_power() {
```

Comment:
surely there's a clearer way to do all this Option wrangling?

Reply:
Yes, you're right. Just pushed something that I think works better.